packages/gatsby/src/bootstrap/load-themes/index.ts

```diff
@@ -76,6 +90,21 @@ const resolveTheme = async (
   }
 }
 
+interface IProcessThemeOptions {
+  useLegacyThemes: boolean
+  rootDir: string
+}
+
+interface IProcessedTheme {
+  themeName: string
+  themeConfig: GatsbyConfig
+  themeSpec: PluginRef
+  themeDir: string
+  parentDir: string
+}
+
+type ProcessedThemes = Array<IProcessedTheme | ProcessedThemes>
```

Comment:
Using recursion here. I imagine the user will almost always want to flatten this type. If not clear enough, should I leave a comment?

Reply:
I think this is fine